libs/base/console.ts

```diff
@@ -71,6 +71,46 @@ namespace console {
         log(name ? `${name}: ${value}` : `${value}`)
     }
 
+    /**
+     * Convert any object or value to a string representation
+     * @param obj value to be converted to a string
+     * @param maxElements [optional] max number values in an object to include in output
+     */
+    export function inspect(obj: any, maxElements = 20): string {
+        if (typeof obj == "string") {
+            return obj;
+        } else if (typeof obj == "number") {
+            return "" + obj;
+        } else if (Array.isArray(obj)) {
+            const asArr = (obj as Array<string>);
+            if (asArr.length <= maxElements) {
+                return asArr.join(",");
+            } else {
+                return `${asArr.slice(0, maxElements).join(",")}...`;
+            }
+        } else {
+            const asString = obj + "";
+            if (asString != "[object Object]"
+                && asString != "[Object]") { // on arcade at least, default toString is [Object] on hardware instead of standard
+                return asString;
+            }
+
+            let keys = Object.keys(obj);
+            const snipped = keys.length > maxElements;
+            if (snipped) {
+                keys = keys.slice(0, maxElements);
+            }
+
+            return `{${
+                keys.reduce(
+                    (prev, currKey) => prev + `\n    ${currKey}: ${obj[currKey]}`,
```

Comment:
call inspect again?

Reply:
Currently, any member of the object without a toString will end up as `[object Object]`. I didn't want to make it recursive right away as it seems like it might be too much info to show at once. I think it works nicely in chrome cause you can expand / collapse at will, but non-trivial objects might be really hard to parse if it shows everything right away - I suppose it could just have depth for how far to recurse as a parameter though

Also have to deal with cases like if a node containing a reference back to it's parent if we made it recurse.

Not ideal of course, can look into making this more descriptive - for now I figured showing the  first 'layer' of properties is helpful for giving quick info on what's in the object

Reply:
I think this PR is an improvement over current state. I guess the real solution would need to keep track of the length of the string so far and stop printing when it exceeds some limits, but this is generally tricky. I would say good enough for now